### ISSUE TITLE
Use recommended caching settings for awesome-typescript-loader

### ISSUE THREAD

**Issue by flut1** [hjeti/vue-skeleton#95]
See: https://github.com/s-panferov/awesome-typescript-loader#performance-issues

These settings should speed up compilation. (we should only enable caching for development)

**Comment by flut1** [comment]
@hjeti  I will take care of this if you like. Do you happen to know why we have `node_modules` in the "include" array of `tsconfig.json`? This will most likely slow down compilation a lot. These are the only files you would want in that includes array:

 - The entry point. So in our case that would be bootstrap.ts
 - Any files that aren't explicitly referenced in your code, but that still need to be included for typings. For example: a `definitions.d.ts` file.

While I don't think that it matters much to include our entire `src` folder (we probably use all these typings anyway), there are potentially a lot of .ts files in node_modules that are irrelevant to our projects.

**Comment by ThaNarie** [comment]
I believe WebStorm likes to have the full `src/**/*.ts` in there to enable project-wide type checking, it doesn't follow imports itself.

**Comment by flut1** [comment]
Yeah I think what I said about the entry point might be true for the `files` array rather than `include`. Regardless I agree that we should keep `"src/**/*.ts"`. In my experience it's really annoying if Webstorm doesn't pick up your new TS files unless you import them from somewhere.
That does still leave `node_modules` however, which we should probably remove.

**Comment by flut1** [comment]
I ran into the following issue on a different project: s-panferov/awesome-typescript-loader#528

Will need to investigate either adding fork-ts-checker plugin like the comments suggest, or maybe look into using ts-loader instead

**Comment by flut1** [comment]
I think I'll have to try both ts-loader, awesome-typescript-loader and babel-typescript in 1 or more projects to see which is fastest. I'll do some benchmarks and will let you know. Additional caching like hard-source-cache-loader seems to make a huge difference in all 3 of these.
